ibtk/src/math/SAMRAIGhostDataAccumulator.cpp

```diff
@@ -124,6 +111,8 @@ add_or_get(Pointer<PatchLevel<NDIM> >& level,
         IBTK_CHKERRQ(ierr);
         for (unsigned int d = 0; d < dofs_ptrs.size(); ++d)
         {
+            const PetscInt size = dofs_ptrs[d]->getDepth() * dofs_ptrs[d]->getOffset();
```

Comment:
This is the part that replaces the box usage - `getDepth() * getOffset()` is exactly the size of the array.